src/platform/packages/shared/shared-ux/chrome/navigation/src/ui/components/panel/default_content.tsx

```diff
@@ -52,17 +48,6 @@ function serializeChildren(node: PanelSelectedNode): ChromeProjectNavigationNode
     ];
   }
 
-  const allChildrenAreGroups = node.children.every((_node) => {
-    if (_node.renderAs === 'item') return false;
-    return isGroupNode(_node);
-  });
-
-  if (!allChildrenAreGroups) {
-    throw new Error(
-      `[Chrome navigation] Error in node [${node.id}]. Children must either all be "groups" or all "items" but not a mix of both.`
-    );
-  }
-
```

Comment:
This code is also fixed in https://github.com/elastic/kibana/pull/216109, which provides a unit test for the fix.